src/main/java/net/neoforged/neoforge/common/util/NeoForgeExtraCodecs.java

```diff
@@ -315,4 +317,44 @@ public String toString() {
             return "XorMapCodec[" + first + ", " + second + "]";
         }
     }
+
+    /**
+     * Creates a codec for an {@linkplain Codec#unboundedMap(Codec, Codec) unbounded map} whose underlying representation is a list of maps, with the given names for each key-element entry. Each key-element entry is encoded as a map with the given key and element names respectively.
+     * <p>
+     * This is useful for maps where the key does not encode to a string, which causes errors when trying to serialize to a format that requires maps to have string keys (such as JSON and NBT).
+     *
+     * @param keyName      the name of the key in the encoded map for each key-element entry
+     * @param keyCodec     codec for the key
+     * @param elementName  the name of the element in the encoded map for each key-element entry
+     * @param elementCodec codec for the element
+     * @param <K>          the key type
+     * @param <V>          the element type
+     * @return a codec for an unbounded map
+     */
+    public static <K, V> Codec<Map<K, V>> unboundedMapAsList(String keyName, Codec<K> keyCodec, String elementName, Codec<V> elementCodec) {
+        Codec<Pair<K, V>> pairCodec = RecordCodecBuilder.create(
```

Comment:
This is not blocking, but you could use `Map.Entry` directly and then `Map#ofEntries` to avoid the streams.